### PR TITLE
Export full Cocina json

### LINKED ISSUE #3847: Add bulk action to get full object cocina
Sometimes you need the full cocina for a set of objects in the original JSON because the flattening effect of CSV transformations makes a CSV version too hard to parse.

### PR DESCRIPTION
**Pull request by lwrubel**
# Why was this change made? 🤔
Resolves #3847. Download Cocina JSON

# How was this change tested? 🤨
Unit, tested on QA. Checking with @andrewjbtw about whether he would like to test that. 

# Does your change introduce accessibility violations? 🩺
No

### CODE REVIEW

File: app/jobs/export_cocina_json_job.rb [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+# @param [Hash] params additional parameters that an Argo job may need
+class ExportCocinaJsonJob < GenericJob
+  def perform(bulk_action_id, params)
+    super
+
+    with_bulk_action_log do |log|
+      File.open(json_download_path, "w") do |jsonl_file|
+        with_items(params[:druids], name: "Export cocina JSON") do |cocina_object, success, failure|
+          if cocina_object.is_a?(NilModel)
+            failure.call("No JSON exported since not object not found (bulk_action.id=#{bulk_action_id})")
```
Comment (justinlittman) [inline app/jobs/export_cocina_json_job.rb:15]: "No JSON exported since not object not found" <-- Wording off.

**Review by justinlittman — APPROVED** [review]
Approve pending fix of typo.

**Comment by lwrubel** [comment]
Fixed the linting issue thanks to a pointer from @justinlittman.

**Comment by mjgiarlo** [comment]
@lwrubel 💬 
> ...
> I am also stumped how to deal with this linting error:
> 
> ```
> spec/jobs/export_cocina_json_job_spec.rb:27:5: C: RSpec/SubjectStub: Do not stub methods of the object under test.
>     allow(job).to receive(:bulk_action).and_return(bulk_action)
>     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
> ```

Maybe I'm alone in this, but I tend not to care too much about this rule when the test is clear and useful, and I'd say that's true in this case. Maybe a good candidate for `# rubocop:disable RSpec/SubjectStub`

File: app/jobs/export_cocina_json_job.rb [hunk 2]
```diff
@@ -0,0 +1,36 @@
+# frozen_string_literal: true
+
+##
+# A job that exports line-oriented JSON for one or more objects
+# @param [Integer] bulk_action_id GlobalID for a BulkAction object
+# @param [Hash] params additional parameters that an Argo job may need
```
Comment (mjgiarlo) [inline app/jobs/export_cocina_json_job.rb:6]: We might want these two lines immediately above `def perform` rather than on the class def

File: app/jobs/export_cocina_json_job.rb [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+class ExportCocinaJsonJob < GenericJob
+  def perform(bulk_action_id, params)
+    super
+
+    File.open(json_download_path, "w") do |jsonl_file|
+      with_items(params[:druids], name: "Export cocina JSON") do |cocina_object, success, failure|
```
Comment (mjgiarlo) [inline app/jobs/export_cocina_json_job.rb:12]: I can't remember how the `name` param is used, so maybe it's not worth doing here, but `cocina` is capitalized most other strings.

Comment (lwrubel) [reply app/jobs/export_cocina_json_job.rb:12]: It shows up in the bulk action's log, mainly for failures, e.g. `2023-08-30 17:33:17 UTC Export cocina JSON failed Dor::Services::Client::NotFoundResponse ...`

Agree it's more consistent to capitalize, so doing that, thanks!
